Dxc fix verifyrootsignature option

yskim1501 opened the pull request:
- distinguish stream type (stdout, stderr) when printing to console
- verifyrootsignature option returns correct errorlevel
- adding more test cases to hcttest for verifyrootsignature option
- enable verifyrootsignature option when dxil.dll not present

yskim1501 committed:
Dxc fix verifyrootsignature option

- distinguish stream type (stdout, stderr) when printing to console
- verifyrootsignature option returns correct errorlevel depending on its result
- adding more test cases to hcttest for verifyrootsignature option
- enable verifyrootsignature option when dxil.dll not present

tex3d commented on include/dxc/Support/dxcapi.use.h:
```diff
@@ -104,13 +104,13 @@ void IFT_Data(HRESULT hr, _In_opt_ LPCWSTR data);
 void ReadFileIntoBlob(DxcDllSupport &dxcSupport, _In_ LPCWSTR pFileName,
                       _Outptr_ IDxcBlobEncoding **ppBlobEncoding);
-void WriteBlobToConsole(_In_opt_ IDxcBlob *pBlob);
+void WriteBlobToConsole(_In_opt_ IDxcBlob *pBlob, FILE* streamType = stdout);
```
See other API's use of HANDLE for file handles.  Should we be using that instead?  You can use GetStdHandle to get the corresponding handle for stdout/stderr.

tex3d commented on tools/clang/tools/dxcompiler/dxcontainerbuilder.cpp:
```diff
@@ -195,7 +233,7 @@ HRESULT CreateDxcContainerBuilder(_In_ REFIID riid, _Out_ LPVOID *ppv) {
   HRESULT hr = DxilLibCreateInstance(CLSID_DxcContainerBuilder, (IDxcContainerBuilder**)ppv);
   if (FAILED(hr)) {
-    warning = "Unable to create container builder from dxil.dll, fallback to built-in.";
+    warning = "Unable to create container builder from dxil.dll, fallback to built-in.\n";
```
Perhaps instead of "fallback to built-in" we should say "resulting container will not be signed".  This puts it in the perspective of "what does this mean" for the customer.

yskim1501 committed:
Fixes from review

- Change parameters for writing to consoles from File* to DWORD
- Change comment when dxil.dll does not exist
